src/components/Collaborative/Form/SourceExtesion.tsx

```diff
@@ -0,0 +1,66 @@
+import {
+    DOMCompatibleAttributes,
+    ExtensionTag,
+    NodeExtension,
+    NodeExtensionSpec,
+} from "@remirror/core";
+import React, { ComponentType } from "react";
+import { NodeViewComponentProps } from "@remirror/react";
+import { SourceCard } from "./SourceCard";
+
+class SourceExtension extends NodeExtension {
+    get name() {
+        return "source" as const;
+    }
+
+    ReactComponent: ComponentType<NodeViewComponentProps> = ({
+        forwardRef,
+    }) => {
+        return <SourceCard forwardRef={forwardRef} />;
+    };
+
+    createTags() {
+        return [ExtensionTag.Block];
+    }
+
+    createNodeSpec(): NodeExtensionSpec {
+        return {
+            selectable: false,
```

Comment:
suggestion: create a `BaseNodeExtensionSpec` to avoid copying and pasting most of the properties we are always setting.